pkg/controller/nodes/array/execution_context.go

```diff
@@ -0,0 +1,48 @@
+package array
```

Comment:
how about a comment explaining why we need a special context. As i do not think we have a special context in any other node type?

Reply:
great idea, added it on the `buildArrayNodeContext` function in the handler code.